Content.Client/UserInterface/Controls/RadialContainer.cs

```diff
@@ -67,36 +67,86 @@ public RadialContainer()
+        arc = arc < 0
+            ? MathF.Tau + arc
+            : arc;
+        arc = isAntiClockwise
+            ? MathF.Tau - arc
+            : arc;
 
         // Account for both circular arrangements and arc-based arrangements
-        var childMod = MathHelper.CloseTo(arc, MathF.Tau, 0.01f) ? 0 : 1;
+        var childMod = MathHelper.CloseTo(arc, MathF.Tau, 0.01f)
+            ? 0
+            : 1;
 
         // Determine the separation between child elements
-        var sepAngle = arc / (childCount - childMod);
-        sepAngle *= (RadialAlignment == RAlignment.AntiClockwise) ? -1f : 1f;
+        var sepAngle = arc / (elementCount - childMod);
+        sepAngle *= isAntiClockwise
+            ? -1f
+            : 1f;
+
+        var halfWidth = Width / 2f;
+        var halfHeight = Height / 2f;
 
         // Adjust the positions of all the child elements
-        foreach (var (i, child) in children.Select((x, i) => (i, x)))
+        var query = children.Select((x, index) => (index, x));
+
+        foreach (var (index, child) in query)
         {
-            var position = new Vector2(Radius * MathF.Sin(AngularRange.X + sepAngle * i) + Width / 2f - child.Width / 2f, -Radius * MathF.Cos(AngularRange.X + sepAngle * i) + Height / 2f - child.Height / 2f);
+            // odd indexes are child elements (buttons)
+            var indexForChildElement = index * 2 + 1;
+
+            var targetAngleOfChild = AngularRange.X + sepAngle * indexForChildElement;
+            var childX = Radius * MathF.Sin(targetAngleOfChild) + halfWidth - child.Width / 2f;
+            var childY = -Radius * MathF.Cos(targetAngleOfChild) + halfHeight - child.Height / 2f;
+            var position = new Vector2(childX, childY);
+
             SetPosition(child, position);
+
+            // radial menu buttons need to know in which sector around container they should be rendered
+            if (child is RadialMenuTextureButton tb)
+            {
+                tb.AngleSectorFrom = sepAngle * (indexForChildElement - 1);
+                tb.AngleSectorTo = sepAngle * (indexForChildElement + 1);
+            }
+
+            // draw separator lines for buttons
+            var positionOfSeparator = index * 2;
+
+            var targetAngleOfSeparator = AngularRange.X + sepAngle * positionOfSeparator;
+            var separatorSin = MathF.Sin(targetAngleOfSeparator);
+            var separatorCos = MathF.Cos(targetAngleOfSeparator);
+            handle.DrawLine(
+                new Vector2(
+                    Radius/2 * separatorSin + halfWidth,
+                    -Radius/2 * separatorCos + halfHeight
+                ) * UIScale,
+                new Vector2(
+                    Radius * 2 * separatorSin + halfWidth,
+                    -Radius * 2 * separatorCos + halfHeight
+                ) * UIScale,
+                new Color(173, 216, 230, 180) // todo: use stylesheets
```

Comment:
this is not button but i'd like to use styles from buttons! Buttom background color when its hovered, to be precise! I don't really understand how i can do it - TryGetStyleProperty is working in a wierd way even when used in proper class hierarchy, and here it is quite empty Q_Q

Reply:
The colors can be defined as separate fields instead of styles

Reply:
Yea, this PR is getting bloated, we will have to just leave it as fields with default values, i'm not going for stylesheets. Sadge.